packages/library-base/src/commonMain/kotlin/io/realm/internal/BaseResults.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal
+
+import io.realm.RealmObject
+import io.realm.RealmResults
+import io.realm.internal.interop.Callback
+import io.realm.internal.interop.Link
+import io.realm.internal.interop.NativePointer
+import io.realm.internal.interop.RealmCoreException
+import io.realm.internal.interop.RealmInterop
+import kotlinx.coroutines.channels.ChannelResult
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.flow.Flow
+import kotlin.reflect.KClass
+
+/**
+ * This class gathers all type-agnostic logic for handling [RealmResults]. This class is needed as
+ * Core can produce both objects and primitive results.
```

Comment:
It appears to be true for dictionaries. However, Claus and I talked to Jørgen about it and he sounded as though Core returning results is a "misuse of the results API". It may be necessary to spark a conversation about this with the other SDKs to see what alternatives there are to avoid exposing scalar results.

Reply:
Furthermore, I think leaving scalar results out is perfectly reasonable for the time being. We can perfectly manage using ordinary `RealmResults<RealmObject>` for now to compute query aggregators.